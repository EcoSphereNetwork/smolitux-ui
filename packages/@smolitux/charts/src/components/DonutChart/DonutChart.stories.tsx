import React from 'react';
import { Meta, StoryObj } from '@storybook/react';
import { DonutChart } from './DonutChart';

const meta: Meta<typeof DonutChart> = {
  title: 'Charts/DonutChart',
  component: DonutChart,
  parameters: { layout: 'centered' },
  tags: ['autodocs'],
  argTypes: {
    height: { control: { type: 'number' } },
    width: { control: { type: 'text' } },
    showLegend: { control: { type: 'boolean' } },
    showValues: { control: { type: 'boolean' } },
    legendPosition: {
      control: { type: 'radio' },
      options: ['top', 'right', 'bottom', 'left'],
    },
  },
};
export default meta;

type Story = StoryObj<typeof DonutChart>;

const sampleData = [
  { label: 'A', value: 40 },
  { label: 'B', value: 60 },
];

export const Default: Story = {
  args: {
    data: sampleData,
    height: 300,
    width: 300,
    title: 'Distribution',
    showLegend: true,
    showValues: true,
  },
};
