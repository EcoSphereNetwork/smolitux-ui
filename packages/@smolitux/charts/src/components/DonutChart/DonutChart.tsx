import React, { forwardRef } from 'react';
import PieChart, { type PieChartProps } from '../PieChart/PieChart';

export interface DonutChartProps extends Omit<PieChartProps, 'donut'> {}

/**
 * DonutChart is a thin wrapper around PieChart with donut mode enabled by default.
 */
export const DonutChart = forwardRef<SVGSVGElement, DonutChartProps>(({ ...props }, ref) => (
  <PieChart ref={ref} donut {...props} />
));

DonutChart.displayName = 'DonutChart';

export default DonutChart;
