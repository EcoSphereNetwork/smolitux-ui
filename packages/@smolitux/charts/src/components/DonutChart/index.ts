export { default, type DonutChartProps } from './DonutChart';
