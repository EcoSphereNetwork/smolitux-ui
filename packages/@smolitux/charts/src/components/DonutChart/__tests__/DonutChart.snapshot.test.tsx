import React from 'react';
import { render } from '@testing-library/react';
import { DonutChart } from '../DonutChart';

jest.mock('@smolitux/theme', () => ({
  useTheme: () => ({ themeMode: 'light' }),
}));

describe('DonutChart Snapshots', () => {
  const mockData = [
    { label: 'A', value: 40 },
    { label: 'B', value: 60 },
  ];

  it('renders default chart', () => {
    const { asFragment } = render(<DonutChart data={mockData} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
