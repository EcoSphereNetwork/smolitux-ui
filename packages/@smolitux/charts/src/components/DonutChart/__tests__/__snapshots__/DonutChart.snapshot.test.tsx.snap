// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DonutChart Snapshots renders default chart 1`] = `
<DocumentFragment>
  <div
    class="w-full "
    style="display: flex; flex-direction: column; align-items: center; width: 100%; height: 100%;"
  >
    <div
      style="order: 1;"
    >
      <svg
        class="smolitux-pie-chart "
        height="300"
        preserveAspectRatio="xMidYMid meet"
        role="img"
        viewBox="0 0 100 100"
        width="100%"
      >
        <g>
          <g>
            <path
              class="
                    transition-opacity duration-300
                    opacity-100
                    
                  "
              d="
          M 50 15
          A 35 35 0 0 1 70.57248383023656 78.31559480312316
          L 58.22899353209462 61.326237921249266
          A 14 14 0 0 0 50 36
          Z
        "
              fill="#3B82F6"
              stroke="white"
              stroke-width="0.5"
            >
              <animate
                attributeName="opacity"
                begin="0s"
                dur="0.8s"
                fill="freeze"
                from="0"
                to="1"
              />
            </path>
          </g>
          <g>
            <path
              class="
                    transition-opacity duration-300
                    opacity-100
                    
                  "
              d="
          M 70.57248383023656 78.31559480312316
          A 35 35 0 1 1 49.99999999999999 15
          L 50 36
          A 14 14 0 1 0 58.22899353209462 61.326237921249266
          Z
        "
              fill="#10B981"
              stroke="white"
              stroke-width="0.5"
            >
              <animate
                attributeName="opacity"
                begin="0.1s"
                dur="0.8s"
                fill="freeze"
                from="0"
                to="1"
              />
            </path>
          </g>
        </g>
        <circle
          cx="50"
          cy="50"
          fill="white"
          r="14"
        />
        <g>
          <text
            class="fill-gray-900 dark:fill-white text-sm font-bold"
            dominant-baseline="middle"
            style="font-size: 4px;"
            text-anchor="middle"
            x="50"
            y="48"
          >
            100
          </text>
        </g>
      </svg>
    </div>
    <div
      style="display: flex; flex-direction: column; order: 2;"
    >
      <div
        class="mt-4 w-full"
      >
        <div
          class="flex flex-wrap"
        >
          <div
            class="flex items-center mb-1 cursor-pointer "
          >
            <div
              class="w-3 h-3 mr-2 rounded-sm"
              style="background-color: rgb(59, 130, 246);"
            />
            <div
              class="text-xs"
            >
              <span
                class="legend-label font-medium"
              >
                A
              </span>
              <span
                class="ml-2 text-gray-500 dark:text-gray-400"
              >
                40 (40.0%)
              </span>
            </div>
          </div>
          <div
            class="flex items-center mb-1 cursor-pointer "
          >
            <div
              class="w-3 h-3 mr-2 rounded-sm"
              style="background-color: rgb(16, 185, 129);"
            />
            <div
              class="text-xs"
            >
              <span
                class="legend-label font-medium"
              >
                B
              </span>
              <span
                class="ml-2 text-gray-500 dark:text-gray-400"
              >
                60 (60.0%)
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <style>
      
        @keyframes fillIn {
          from { opacity: 0; }
          to { opacity: 1; }
        }
      
    </style>
  </div>
</DocumentFragment>
`;
